### PR TITLE
Upgrade to Expo v42

### PR DESCRIPTION
**Pull request by frxnz**
Upgrade from Expo 41 to 42

Release docs: https://blog.expo.io/expo-sdk-42-579aee2348b6

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,21 +18,21 @@
...
     "expo-app-loading": "^1.0.3",
-    "expo-font": "~9.1.0",
-    "expo-status-bar": "^1.0.4",
+    "expo-font": "~9.2.1",
+    "expo-status-bar": "~1.0.4",
     "history": "^5.0.0",
     "immer": "^9.0.1",
     "normalizr": "^3.6.1",
     "prop-types": "^15.7.2",
-    "react": "^17.0.2",
```
Comment (frxnz) [inline package.json:29]: Expo officially supports React 16, and most plugins list it as a peer dependency. So for now let's just use React 16.

Comment (wswoodruff) [reply package.json:29]: Broooonnnnnkyyyyyyyy!!!!

File: package.json [hunk 2]
```diff
@@ -18,21 +18,21 @@
     "@react-navigation/native": "^5.9.4",
     "@react-navigation/stack": "^5.14.4",
     "envy": "^2.0.0",
-    "expo": "^41.0.0",
+    "expo": "^42.0.0",
     "expo-app-loading": "^1.0.3",
-    "expo-font": "~9.1.0",
-    "expo-status-bar": "^1.0.4",
+    "expo-font": "~9.2.1",
```
Comment (devinivy) [inline package.json:23]: Are we using the tilde because these are big/finicky dependencies?

File: package.json [hunk 3]
```diff
@@ -18,21 +18,21 @@
...
+    "expo-status-bar": "~1.0.4",
     "history": "^5.0.0",
     "immer": "^9.0.1",
     "normalizr": "^3.6.1",
     "prop-types": "^15.7.2",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "react-native": "https://github.com/expo/react-native/archive/sdk-41.0.0.tar.gz",
+    "react": "16.13.1",
+    "react-dom": "16.13.1",
```
Comment (devinivy) [inline package.json:30]: Any special reason that we pin these?  As I see it the package-lock will/can pin them for us on a per-project basis.

File: package.json (hunk 2 above)
Comment (frxnz) [reply package.json:23]: Expo CLI did this, and I trust it more than I trust me!

File: package.json (hunk 3 above)
Comment (frxnz) [reply package.json:30]: Expo CLI did this, but Expo/React-Native expects a very specific version of React so this is a nice indication that you shouldn't mess with these.

**Comment by devinivy** [comment]
How would we feel about omitting the package-lock file?  I find it's common practice for boilerplates to generate the package-lock during installation (strangeluv, pal, create-react-app), I expect so that the user is getting up-to-date dependencies and so that the boilerplate doesn't have to be responsible for maintaining deps as granularly.

File: package.json (hunk 2 above)
Comment (devinivy) [reply package.json:23]: Ah, got it!  Yes, happy to pass off that job to expo :P

File: package.json (hunk 3 above)
Comment (devinivy) [reply package.json:30]: Ah, didn't realize!  Sounds good 👍

**Comment by frxnz** [comment]
@devinivy Consider it done!

File: .gitignore [hunk 4]
```diff
@@ -117,4 +117,6 @@ dist
 
 .expo
 
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+package-lock.json
```
Comment (devinivy) [inline .gitignore:122]: At the same time we probably don't want the application itself to ignore package-lock— I think we just want to let them generate it for themselves.

Comment (frxnz) [reply .gitignore:122]: shrexpo's own .gitignore isn't consumed on installation, so this _should_ be chill. If there's a better way though lmk!

Comment (devinivy) [reply .gitignore:122]: Ah got it, excellent 👍 I think that is ideal.

**Review by devinivy — APPROVED** [review]

**Pull request merged**